server/src/main/java/org/elasticsearch/snapshots/RestoreService.java

```diff
@@ -988,6 +991,19 @@ private static void validateSnapshotRestorable(final RepositoryMetadata reposito
         }
     }
 
+    static void maybeFixRestoreGlobalStateFlag(RestoreSnapshotRequest request, Snapshot snapshot, SnapshotInfo snapshotInfo) {
+        if (request.includeGlobalState() && snapshotInfo.includeGlobalState() != Boolean.TRUE) {
+            request.includeGlobalState(false);
+            logger.warn(
```

Comment:
Should we rather fail out here? Seems like an inadvertent use of a snapshot?

Reply:
@henningandersen yes me and @idegtiarenko were one the same page on this as well but figured we'd have to ask you here since it's technically an API breaking change? :) If you're good with that kind of change, ++ I think we should just straight opt for failing.